apps/server/src/modules/common-cartridge/utils.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { InternalServerErrorException } from '@nestjs/common';
+import { ObjectID } from 'bson';
+import { CommonCartridgeVersion } from './common-cartridge.enums';
+import {
+	buildXmlString,
+	checkDefined,
+	checkIntendedUse,
+	createElementTypeNotSupportedError,
+	createIdentifier,
+	createResourceTypeNotSupportedError,
+	createVersionNotSupportedError,
+} from './utils';
+
+describe('CommonCartridgeUtils', () => {
+	describe('buildXmlString', () => {
+		it('should create xml string', () => {
+			const xml = buildXmlString({ root: { child: 'value' } });
+
+			expect(xml).toBe('<?xml version="1.0" encoding="UTF-8"?>\n<root>\n    <child>value</child>\n</root>');
+		});
+	});
+
+	describe('createVersionNotSupportedError', () => {
+		describe('when creating error', () => {
+			it('should return error with message', () => {
+				const error = createVersionNotSupportedError(CommonCartridgeVersion.V_1_0_0);
+
+				expect(error).toBeInstanceOf(InternalServerErrorException);
+				expect(error.message).toBe('Common Cartridge version 1.0.0 is not supported');
+			});
+		});
+	});
+
+	// AI next 16 lines
+	describe('checkDefined', () => {
+		describe('when checking value', () => {
+			it('should return value when value is defined', () => {
+				const value = 'value';
+
+				expect(checkDefined(value, 'value')).toBe(value);
+			});
+
+			it('should throw error when value is undefined', () => {
+				expect(() => checkDefined(undefined, 'value')).toThrow(new Error('value is null or undefined'));
+			});
+
+			it('should throw error when value is null', () => {
+				expect(() => checkDefined(null, 'value')).toThrow(new Error('value is null or undefined'));
+			});
+		});
+	});
+
+	describe('checkIntendedUse', () => {
+		describe('when checking supported intended use', () => {
+			it('should not throw', () => {
+				const intendedUse = 'unspecified';
+
+				expect(() => checkIntendedUse(intendedUse, ['unspecified'])).not.toThrow();
+			});
+
+			it('should throw error when intended use is not supported', () => {
+				const intendedUse = 'unsupported';
+
+				expect(() => checkIntendedUse(intendedUse, [])).toThrow(`Intended use ${intendedUse} is not supported`);
+			});
+		});
+	});
+
+	describe('createIdentifier', () => {
+		describe('when creating identifier', () => {
+			it('should return identifier with prefix', () => {
+				const identifier = new ObjectID();
+
+				expect(createIdentifier(identifier)).toBe(`i${identifier.toHexString()}`);
```

Comment:
I am unsure if this util make sense. It only wrap the toHexString() operation. But not the bson package or the objectId it self. I think the code is better understandable if you use directly the .toHexString() method instead of createIdentifier.
In the end the util do the same like the source:
```
const a = new ObjectId();
const b = new ObjectId(a);
b.toHexString()

const a = undefined;
const b = new ObjectId(a);
b.toHexString()
```

Reply:
A hex string is not a valid ID, because it can start with an number and the Common Cartridge specification does not allow this.